apache/yunikorn-k8shim
# [YUNIKORN-2760] `make tools` should check the version of tools

### What is this PR for?
Makefile, by default, checks only the existence of file. Hence, developers need to remove tools folder (or call `make distclean`) manually to trigger the installation after we update the version of tools.

However, how developers can be aware of the tools updates? Personally, I smell fishy from the error of warning, but that could be implicit and noisy :cry

In order to fix that, I'd like to introduce the new folder structure to tools folder: 

```
/tools/{tool_name}-{version}
```

That offers a unique path to each version of tool. Developers will not miss the updates anymore.


### What type of PR is it?

### What is the Jira issue?
[YUNIKORN-2760](https://issues.apache.org/jira/projects/YUNIKORN/issues/YUNIKORN-2760)

### How should this be tested?

Currently, `make tools` is triggered only by `./scripts/run-e2e-tests.sh`
To verify whether `make tools` checks the tools' versions, call any command in `./scripts/run-e2e-tests.sh` or trigger it manually

### Screenshots (if appropriate)

![image](https://github.com/user-attachments/assets/db1dab6c-9022-43ad-868f-b1a044f49573)
